ingest/indexer/service/indexer_streaming_service.go

```diff
@@ -340,13 +345,23 @@ func (s *indexerStreamingService) ListenCommit(ctx context.Context, res abci.Res
 	s.blockUpdatesProcessUtils.SetChangeSet(changeSet)
 
 	// Create block processor
-	blockProcessor := blockprocessor.NewBlockProcessor(s.blockProcessStrategyManager, s.client, s.poolExtractor, s.keepers, s.blockUpdatesProcessUtils)
+	// Note the returned block processor can be either full or incremental depending on the strategy
+	// When node is syncing, it will be a full block processor
+	// When node is already synced, it will be an incremental block processor
+	blockProcessor := blockprocessor.NewBlockProcessor(s.blockProcessStrategyManager, s.client, s.poolExtractor, s.keepers, s.nodeStatusChecker, s.blockUpdatesProcessUtils)
 
 	// Process block.
 	if err := blockProcessor.ProcessBlock(sdkCtx); err != nil {
+		// In the case of full block processor, if any error is returned, including node is syncing or sync check fails,
+		// data is not marked as ingested and will be retried in the next block
 		return err
 	}
 
+	// If block processor is a full block processor, mark the initial data as ingested
+	if blockProcessor.IsFullBlockProcessor() {
+		s.blockProcessStrategyManager.MarkInitialDataIngested()
+	}
+
```

Comment:
Only when no error is thrown in the above (which implies node sync is no longer in progress and full block processor has completed successfully), we mark the data as ingested by invoking `MarkInitialDataIngested`, so that in the next block, an incremental block processor (not full block process) will be used.